keymapper/injection/consumers/consumer.py

```diff
@@ -54,12 +59,21 @@ def is_enabled(self):
 
     def write(self, key):
         """Shorthand to write stuff."""
-        self.context.uinput.write(*key)
-        self.context.uinput.syn()
+        uinput = global_uinputs.get_appropriate_uinput(key)
+        if uinput is None:
+            uinput = self.context.miscellaneous_output
+
+        uinput.write(*key)
+        uinput.syn()
 
     def forward(self, key):
         """Shorthand to forward an event."""
-        self.forward_to.write(*key)
+        uinput = global_uinputs.get_appropriate_uinput(key)
+        if uinput is None:
+            uinput = self.forward_to
```

Comment:
I am not sure if we actually want to catch every possible event with a global UInput as long as a single event of that group is mapped. This seems quite confusing. I would expect any event which is not mapped to end up in the forwarded device.

Reply:
I see

For a mapping of `shift_l + a: b` we definitely have to write all shift_l and a events to the global_uinputs, so that `e(EV_KEY, KEY_LEFTSHIFT, 0)` can be used to release the shift.